### PR TITLE
Proposed UX articles

### PR DESCRIPTION
**Pull request by jncrabb**
## Description

Two articles covering some basic Web3 UX to help all builders in the Ethereum ecosystem:
- A list of heuristics to use when assessing usability of a dApp
- A guide to the construction of a DEX, or any component where you swap tokens

The guide to DEX will also feature a Figma wireframe file, although this will need to be uploaded and shared first, ideally to an official account.

## Preview URL
- https://deploy-preview-13075--ethereumorg.netlify.app/en/developers/docs/design-and-ux/heuristics-for-web3/

### CODE REVIEW

**Comment by wackerow** [comment]
Hey @jncrabb, thanks for proposing this! At a glance looks like it'd be a nice addition. @konopkja @nloureiro Mind taking a look through this for content review?

**Comment by konopkja** [comment]
content LGTM! will probably need dev to fix few issues such as missing name in the docs navigation and double header

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+Arguably, some of these details could be optional.
+
+Order routing is interesting, but doesn’t make much difference to most users.
+
+Some other details are simply restating the same thing in different ways. For example “minimum received” and “slippage” are two sides of the same coin. If you have slippage set at 1%, then the minimum you can expect to receive = expected output-1%. Some UIs will show expected amount, minimum amount, and slippage… Which is useful but possibly overkill. 
+
+Most users will leave slippage on Auto anyway.
+
+“Price impact” is often shown in brackets next to the usd equivalent in the “to” field. This is a great ux detail to add, but if it is shown here, does it really need to be shown again below? And then again on a preview screen?
```
Comment (corwintines) [inline public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:103]:
```suggestion
“Price impact” is often shown in brackets next to the fiat equivalent in the “to” field. This is a great ux detail to add, but if it is shown here, does it really need to be shown again below? And then again on a preview screen?
```

**Review by corwintines — COMMENTED** [review]
Nice work overall @jncrabb 

I have a few suggestions here, but overall think this is close.

File: public/content/developers/docs/design-and-ux/heuristics-for-web3/index.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+### 5. Actions are as Short as Possible
+
+Speed up the user’s interactions by accelerating and grouping sub actions. 
+This may be done on the smart contract level, as well as the UI. The user should not have to move from one part of the system to another – or leave the system entirely – to complete a common action. 
+
+**Tips:**
+- Incorporate zappers for adding liquidity
+- Combine “Approve” with the “Deposit” action
+- Combine “Deposit” and “Stake” actions
+- Allow the user to close a loan position and withdraw collateral entirely
```
[the remark this answers is not in the document]
Comment (jncrabb) [reply public/content/developers/docs/design-and-ux/heuristics-for-web3/index.md:101]: What tips would you suggest? I think it needs to be actionable, so would need a specific example for a non-DeFi app.

[the remark this answers is not in the document]
Comment (corwintines) [reply public/content/developers/docs/design-and-ux/heuristics-for-web3/index.md:101]: **Tips:**
- Combine "Approve" with other actions where possible
- Bundle signing steps as close together as possible

Trying to think on this a bit, this is a start I have but will do some more thinking. I know the second tip could be worded better. The first one is just an iteration on what you have but made a bit more general.

[the remark this answers is not in the document]
Comment (corwintines) [reply public/content/developers/docs/design-and-ux/heuristics-for-web3/index.md:101]: Ill push this up for now and we can build on this. Would like to get these in for next release :)

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+- connect wallet
+- various errors
+
+The button can also be **mapped to the action** that needs to be performed. For example, if the user cannot swap because they are on the wrong network, the button should say “switch to Ethereum”, and when the user clicks on the button, it should switch the network to Ethereum. This speeds up the user flow significantly.
+
+![Key actions being initiated from the main CTA](./16.png)
+
+![Error message shown within the main CTA](./17.png)
+
+## Build your own with this figma file {#build-your-own-with-this-figma-file}
```
Comment (corwintines) [inline public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:208]: Just wondering if we have a link for this Figma file referenced here

**Review by corwintines — COMMENTED** [review]
@jncrabb I just pushed up a bit of cleanup in terms of spelling and setting up headers/titles for the markdown, as well as the translations for the nav. 

All I see left to get this across the finish line is just my comment below about the Figma file. Seems like we reference one here to go to, but I see no link for it. Other than that, I think we are ready to ship this. 

@konopkja let me know if theres anything else you think needs to come in here.

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md (hunk 3 above)
Comment (konopkja) [reply public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:206]: yes we do, ill commit a change

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+Thanks to the hard work of multiple protocols, DEX design has improved a lot. We know what info the user needs, how we should show it, and how to make the flow as smooth as possible.
+Hopefully this article provides a solid overview of the UX principles. 
+
+If you want to experiment, please feel free to use the Figma wireframe kit. It is kept as simple as possible, but has enough flexibility to build the basic structure in various ways.
+
```
Comment (konopkja) [inline public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:212]:
```suggestion
If you want to experiment, please feel free to use the Figma wireframe kit. It is kept as simple as possible, but has enough flexibility to build the basic structure in various ways.

[Figma wireframe kit](https://www.figma.com/community/file/1393606680816807382/dex-wireframes-kit)
```

**Review by konopkja — APPROVED** [review]

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
+---
+title: Decentralized exchange (DEX) best design practices
```
Comment (konopkja) [inline public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:2]:
```suggestion
title: Decentralized exchange (DEX) design best practices
```

File: public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+## Build your own with this figma file {#build-your-own-with-this-figma-file}
+
+Thanks to the hard work of multiple protocols, DEX design has improved a lot. We know what info the user needs, how we should show it, and how to make the flow as smooth as possible.
+Hopefully this article provides a solid overview of the UX principles. 
+
+If you want to experiment, please feel free to use the Figma wireframe kit. It is kept as simple as possible, but has enough flexibility to build the basic structure in various ways.
+
+[Figma wireframe kit](https://www.figma.com/community/file/1393606680816807382/dex-wireframes-kit)
+DeFi will continue to evolve, and there is always room for improvement.
```
Comment (konopkja) [inline public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md:214]:
```suggestion

DeFi will continue to evolve, and there is always room for improvement. 
```

**Review by corwintines — APPROVED** [review]
LGTM, thank you @jncrabb!

**Comment by corwintines** [comment]
@all-contributors can you add @jncrabb for content

**Comment by konopkja** [comment]
Thank you for your contribution @jncrabb and congrats!

**Pull request merged**